geode-connectors/src/test/java/org/apache/geode/connectors/jdbc/internal/SqlStatementFactoryTest.java

```diff
@@ -74,12 +76,69 @@ public void getUpdateSqlString() throws Exception {
 
   @Test
   public void getInsertSqlString() throws Exception {
-    String expectedStatement = String.format("INSERT INTO %s (%s, %s, %s) VALUES (?,?,?)",
-        TABLE_NAME, VALUE_COLUMN_1_NAME, VALUE_COLUMN_2_NAME, KEY_COLUMN_NAME);
+    String expectedStatement = String.format("INSERT INTO %s (%s,%s,%s) VALUES (?,?,?)",
+        TABLE_NAME, VALUE_COLUMN_1_NAME, VALUE_COLUMN_2_NAME, KEY_COLUMN_1_NAME);
 
     String statement = factory.createInsertSqlString(TABLE_NAME, entryColumnData);
 
     assertThat(statement).isEqualTo(expectedStatement);
   }
 
+  @Test
+  public void getInsertSqlStringGivenNoColumns() throws Exception {
+    valueColumnData.clear();
+    keyColumnData.clear();
+
+    String statement = factory.createInsertSqlString(TABLE_NAME, entryColumnData);
+
+    String expectedStatement = String.format("INSERT INTO %s () VALUES ()", TABLE_NAME);
```

Comment:
This is invalid sql statement; do we need to throw an exception while creating the statement? I believe this will throw exception when executed.

Reply:
This may be invalid, but we really don't expect it to ever happen in the real world (i.e. we should always have at least one column). We just found that the older code generated and even more invalid statement (it had an open paren but no close paren). So this unit test is just verifying that in this case we still balance the parens. All we are testing here is that factory code. It being invalid would be checked at a higher level.